internal/controlplane/handlers_repositories.go

```diff
@@ -483,30 +393,68 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 	return out, nil
 }
 
-func (s *Server) deleteRepositoryAndWebhook(
+// TODO: this probably can probably be used elsewhere
+// returns project ID and github client - this flow of code exists in multiple
+// places
+func (s *Server) getProjectIDAndClient(
 	ctx context.Context,
-	repo db.Repository,
-	projectID uuid.UUID,
-	provider db.Provider,
-) error {
-	tx, err := s.store.BeginTransaction()
+	request HasProtoContext,
+) (uuid.UUID, v1.GitHub, error) {
+	projectID := getProjectID(ctx)
+
+	provider, err := getProviderFromRequestOrDefault(ctx, s.store, request, projectID)
 	if err != nil {
-		return status.Errorf(codes.Internal, "error deleting repository")
+		return uuid.Nil, nil, providerError(err)
 	}
-	defer s.store.Rollback(tx)
 
-	qtx := s.store.GetQuerierWithTransaction(tx)
-	if err := qtx.DeleteRepository(ctx, repo.ID); err != nil {
-		return status.Errorf(codes.Internal, "error deleting repository: %v", err)
+	pbOpts := []providers.ProviderBuilderOption{
+		providers.WithProviderMetrics(s.provMt),
+		providers.WithRestClientCache(s.restClientCache),
 	}
 
-	if err := s.deleteWebhookFromRepository(ctx, provider, projectID, repo); err != nil {
+	p, err := providers.GetProviderBuilder(ctx, provider, projectID, s.store, s.cryptoEngine, pbOpts...)
+	if err != nil {
+		return uuid.Nil, nil, status.Errorf(codes.Internal, "cannot get provider builder: %v", err)
+	}
+
+	if !p.Implements(db.ProviderTypeGithub) {
+		return uuid.Nil, nil, util.UserVisibleError(
+			codes.InvalidArgument,
+			"provider %s is not supported for github webhook",
+			p.GetName(),
+		)
+	}
+
+	client, err := p.GetGitHub()
+	if err != nil {
+		return uuid.Nil, nil, status.Errorf(codes.Internal, "error creating github provider: %v", err)
+	}
```

Comment:
Do we need both the `!p.Implements(...)` check _and_ the error handling from `p.GetGitHub`?  I'd hope that the second would return error if unsupported.